backend/utils/dns/route53.js

```diff
@@ -0,0 +1,250 @@
+/**
+ * DNS utilities for AWS Route 53
+ */
+const Route53 = require('aws-sdk/clients/route53')
+
+const { append } = require('../../utils/string')
+const { getLogger } = require('../../utils/logger')
+
+const log = getLogger('utils.dns.route53')
+
+let CACHED_CLIENT
+const DEFAULT_TTL = 60 // 1 minute
+const AWS_API_VERSION = '2013-04-01'
+
+/**
+ * Return a Google Cloud DNS API client
+ *
+ * @returns {DNS}
+ */
+function getClient(credentials) {
```

Comment:
Does this mean changing credentials via the network admin requires a server restart? Perhaps we could memoize it using the JSON.stringified credentials object instead?

Reply:
Guess it does.  That is the same behavior that's in the DNS modules deployed right now.  I can pull this cached logic with this PR if you think it's important or we can create an issue and fix forward.  Might be good to use a hash of the credentials to know when to throw away the cached clients.

Not really sure why you suggest memoizing the creds?  Probably makes more sense to cache the inited client which might include further metadata or even connections(speculating), rather than potentially recreating the client multiple times a session.

Reply:
I meant memoize the getClient function using the hashed credentials as the cache key, similar to what you also suggested. Probably worth making part of this PR as it's a small change.

Reply:
Ah, gotcha.  Yeah, misunderstood.  Will see about it.

Reply:
Done